CELERY_LOCAL_DEVELOPMENT.md

```diff
@@ -0,0 +1,360 @@
+docker-compose up -d redis
+
+# Or start both Redis and Redis Commander
+docker-compose up -d
+```
+
+Redis will be available at `localhost:6379`
+Redis Commander GUI will be available at `http://localhost:8081`
+
+#### Option C: Using Local Redis Installation (Linux)
+```bash
+# Install Redis (Ubuntu/Debian)
+sudo apt-get install redis-server
+
+# Start Redis
+sudo systemctl start redis-server
+
+# Or run directly
+redis-server
+```
+
+### 3. Configure Environment Variables
+
+Create a `.env` file in your project root (if not already present):
+```bash
+# Redis Configuration for Celery
+CELERY_BROKER_URL=redis://localhost:6379/0
+CELERY_RESULT_BACKEND=redis://localhost:6379/0
+
+# Other existing environment variables...
+SECRET_KEY=your-secret-key
+OPENAI_API_KEY=your-openai-api-key
+# ... etc
+```
+
+### 4. Test the Setup
+
+Test that Celery can connect to Redis:
+```bash
+# Test synchronously (doesn't require worker)
+python manage.py test_celery
+
+# Test asynchronously (requires worker running)
+python manage.py test_celery --async
+```
+
+### 5. Start Celery Worker
+
+In a separate terminal, start the Celery worker:
+```bash
+# Basic worker
+celery -A website worker --loglevel=info
+
+# Worker with specific queue (recommended for production)
+celery -A website worker -Q transcription --loglevel=info
+
+# Worker with concurrency control
+celery -A website worker --concurrency=2 --loglevel=info
+```
+
+### 6. Optional: Start Celery Beat (for scheduled tasks)
+
+If you plan to use scheduled tasks:
+```bash
+# In another terminal
+celery -A website beat --loglevel=info
+```
+
+## Usage Examples
+
+### Asynchronous Audio Transcription
+
+```python
+from transcription.services import TranscriptionService
+from nucleus.models import SessionAudio
+
+# Get a SessionAudio instance
+session_audio = SessionAudio.objects.first()
+
+# Create service
+service = TranscriptionService()
+
+# Process asynchronously (recommended for large files)
+async_result = service.process_session_audio_async(
+    session_audio=session_audio,
+    previous_transcript="Previous session content...",
+    session_notes="Session notes...",
+    use_celery=True  # Default
+)
+
+# Get task ID
+print(f"Task ID: {async_result.id}")
+
+# Check if task is ready
+if async_result.ready():
+    result = async_result.get()
+    print(f"Result: {result}")
+else:
+    print("Task is still processing...")
+```
+
+### Direct Task Usage
+
+```python
+from transcription.tasks import process_session_audio_task
+
+# Submit task directly
+result = process_session_audio_task.delay(
+    session_audio_id=1,
+    previous_transcript="",
+    session_notes=""
+)
+
+# Monitor task
+print(f"Task state: {result.state}")
+print(f"Task info: {result.info}")
+```
+
+## Monitoring and Debugging
+
+### Check Redis Connection
+```bash
+# Connect to Redis CLI
+redis-cli
+
+# Test connection
+127.0.0.1:6379> ping
+PONG
+
+# Check if Celery is using Redis
+127.0.0.1:6379> keys celery*
```

Comment:
Let me know what response I should expect here. What is a success response? Is `(empty array)`?